src/Payments/PlaidSDKIframe.res

```diff
@@ -8,122 +8,112 @@ let make = () => {
   let (publishableKey, setPublishableKey) = React.useState(_ => "")
   let (clientSecret, setClientSecret) = React.useState(_ => "")
   let (isForceSync, setIsForceSync) = React.useState(_ => false)
-  let logger = React.useMemo(() => {
-    HyperLogger.make(~source=Elements(Payment), ~clientSecret, ~merchantId=publishableKey)
-  }, (publishableKey, clientSecret))
+  let logger = React.useMemo(
+    () => HyperLogger.make(~source=Elements(Payment), ~clientSecret, ~merchantId=publishableKey),
+    (publishableKey, clientSecret),
+  )
 
-  React.useEffect0(() => {
-    let handle = (ev: Window.event) => {
-      let json = ev.data->safeParse
-      let metaData = json->getDictFromJson->getDictFromDict("metadata")
-      let linkToken = metaData->getString("linkToken", "")
-      if linkToken->String.length > 0 {
-        let pmAuthConnectorArray =
-          metaData
-          ->getArray("pmAuthConnectorArray")
-          ->Array.map(ele => ele->JSON.Decode.string)
+  let handleParentWindowMessage = (ev: Window.event) => {
+    let json = ev.data->safeParse
+    let metaData = json->getDictFromJson->getDictFromDict("metadata")
+    let linkToken = metaData->getString("linkToken", "")
 
-        setLinkToken(_ => linkToken)
-        setPmAuthConnectorsArr(_ => pmAuthConnectorArray)
-        setPublishableKey(_ => metaData->getString("publishableKey", ""))
-        setClientSecret(_ => metaData->getString("clientSecret", ""))
-        setIsForceSync(_ => metaData->getBool("isForceSync", false))
-      }
+    if linkToken->String.length > 0 {
+      setLinkToken(_ => linkToken)
+      setPmAuthConnectorsArr(_ =>
+        metaData->getArray("pmAuthConnectorArray")->Array.map(JSON.Decode.string)
+      )
+      setPublishableKey(_ => metaData->getString("publishableKey", ""))
+      setClientSecret(_ => metaData->getString("clientSecret", ""))
+      setIsForceSync(_ => metaData->getBool("isForceSync", false))
     }
-    Window.addEventListener("message", handle)
+  }
+
+  React.useEffect(() => {
+    Window.addEventListener("message", handleParentWindowMessage)
```

Comment:
if `handleParentWindowMessage` is only used within the useEffect, it makes sense to define the fn within useEffect itself. Redeclaring the fn on each re-render would be useless, since those functions won't be used anyways.